src/app/js/admin/enrichment/sagas.js

```diff
@@ -39,13 +43,33 @@ export function* handleLaunchEnrichment({ payload: enrichment }) {
 }
 
 export function* handleLaunchAllEnrichment() {
-    const enrichmentLaunchAllRequest = yield select(
-        fromUser.getEnrichmentLaunchAllRequest,
-    );
-
-    yield call(fetchSaga, enrichmentLaunchAllRequest);
-
-    return yield put(loadEnrichments());
+    yield put(launchAllEnrichmentStarted());
+
+    try {
+        const enrichmentLaunchAllRequest = yield select(
+            fromUser.getEnrichmentLaunchAllRequest,
+        );
+
+        const { error } = yield call(fetchSaga, enrichmentLaunchAllRequest);
+
+        if (error) {
+            const state = yield select();
+            const polyglot = getP(state);
+
+            if (error.message === 'circular_dependency_error') {
+                toast(polyglot.t('run_all_enrichment_circular_dependency'), {
+                    type: toast.TYPE.ERROR,
+                });
```

Comment:
You should use yield call instead, no function with side effect should get called directly in a saga.
```suggestion
                yield call(
                    toast,
                    polyglot.t('run_all_enrichment_circular_dependency'),
                    { type: toast.TYPE.ERROR }
                );
```